plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -369,9 +346,10 @@ fu_mediatek_scaler_device_open(FuDevice *device, GError **error)
 		return FALSE;
 
 	/* set the target address -- should be safe */
-	if (!fu_mediatek_scaler_ensure_device_address(self,
-						      FU_DDC_I2C_ADDR_DISPLAY_DEVICE >> 1,
-						      error))
+	if (!fu_i2c_device_set_address(FU_I2C_DEVICE(self->i2c_dev),
```

Comment:
Seeing an error - [x.log](https://github.com/user-attachments/files/16602458/x.log)

```
16:04:47.633 GLib-GObject         invalid cast from 'FuUdevDevice' to 'FuI2cDevice'
```

Reply:
Gah, good catch, thanks! This is because `fu_udev_device_get_siblings_with_subsystem()` doesn't create a FuI2cDevice. I've reverted the FuMediatekScalerDevice bits, we can add it back in when FuMediatekScalerDevice subclasses FuI2cDevice and we get a device created by the backend.